genesis/example/member_domain.go

```diff
@@ -0,0 +1,187 @@
+/*
+ *    Copyright 2018 INS Ecosystem
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package example
+
+import (
+	"fmt"
+
+	"github.com/insolar/insolar/genesis/model/class"
+	"github.com/insolar/insolar/genesis/model/domain"
+	"github.com/insolar/insolar/genesis/model/factory"
+	"github.com/insolar/insolar/genesis/model/object"
+	"github.com/insolar/insolar/genesis/model/resolver"
+)
+
+// MemberDomainName is a name for member domain.
+const MemberDomainName = "MemberDomain"
+
+// MemberDomain is a contract that allows to add new members to system.
+type MemberDomain interface {
+	// Base domain implementation.
+	domain.Domain
+	// CreateMember is used to create new member as a child to domain storage.
+	CreateMember() (string, error)
+	// GetMember returns member from its record in domain storage.
+	GetMember(string) (Member, error)
+}
+
+type memberDomain struct {
+	domain.BaseDomain
+	memberFactoryRecord string
+}
+
+// newMemberDomain creates new instance of MemberDomain.
+func newMemberDomain(parent object.Parent) (MemberDomain, error) {
+	if parent == nil {
+		return nil, fmt.Errorf("parent must not be nil")
+	}
+
+	mf, _ := NewMemberFactory(parent).(*memberFactory)
+
+	md := &memberDomain{
+		BaseDomain: *domain.NewBaseDomain(parent, MemberDomainName),
+	}
+	// Add memberFactory as a child
+	record, err := md.AddChild(mf)
+	if err != nil {
+		return nil, err
+	}
+	md.memberFactoryRecord = record
+	return md, nil
+}
+
+// GetClassID returns string representation of MemberDomain's class.
+func (md *memberDomain) GetClassID() string {
+	return class.MemberDomainID
+}
+
+// CreateMember creates new member as a child to domain storage.
+func (md *memberDomain) CreateMember() (string, error) {
+	// Get child by memberFactoryRecord
+	child, err := md.GetChild(md.memberFactoryRecord)
```

Comment:
This should be done using childResolver, GetChild method should not be public anymore